### ISSUE TITLE
clippy::redundant_pub_crate ignores glob export

### ISSUE THREAD

**Issue by ede1998** [rust-lang/rust-clippy#7862]
Lint name:
clippy::redundant_pub_crate

I tried this code:

```rust
mod foo {
    pub(crate) struct Bar;
}

pub use foo::*;
```

I expected to see this happen:
`cargo clippy -- -W clippy::redundant_pub_crate` should not emit a warning for `pub(crate) struct in private module`. The `pub(crate)` visibility prevents `Bar` from becoming part of the public API. The suggestion to just use pub makes `Bar` publicly visible outside of the crate.

Instead, this happened: No `clippy::redundant_pub_crate` warning because `pub(crate)` is different from just `pub` for this example.

### Meta

**Rust version (`rustc -Vv`):**
```
rustc 1.56.0 (09c42c458 2021-10-18)
binary: rustc
commit-hash: 09c42c45858d5f3aedfa670698275303a3d19afa
commit-date: 2021-10-18
host: x86_64-unknown-linux-gnu
release: 1.56.0
LLVM version: 13.0.0
```

**Comment by Antikyth** [comment]
I would like to mention that I have encountered this issue as well, and it's quite annoying because I have been having clippy auto-applying recommended fixes.

This lint is a "machine applicable" lint, which means it is automatically applied.  The problem, of course, is that it can make `pub(crate)` modules public, leaking them into the public API.

My example module layout:
```
lib.rs --- pub use common::*
        |
        |
        |
pub(crate) common
        |
        |
        |\_________________________________________
        |                   |                      \
        |                   |                      |
 pub(crate) mask     pub(crate) atom     pub(crate) attribute

```
The modules at the bottom get turned into `pub`, when they should be `pub(crate)`.  I think if this bug isn't fixed, it should at least not be machine applicable.
